### PR TITLE
Do domain resolution for portal

### PR DESCRIPTION
**Pull request by chiehmin**
iscsiadm support domain resolution (ex: iscsiadm -m discovery -t sendtargets -p iscsi.chiehmin.com).
However, open_iscsi module will try to match the portal with discovered results which will never
matched cause the discovered results use IP to represent node.

This patch do portal DNS resolution first to solve this situation.

Signed-off-by: Chieh-Min Wang <chiehminw@synology.com>

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by Andersson007** [comment]
does the pr change the `portal` parameter behavior? if yes, add info about this to the section
```
    portal:
        description:
        - The IP address of the iSCSI target.
        type: str
        aliases: [ ip ]
```
e.g. `- The IP address or domain name of the iSCSI target.`
also it's maybe worth changing the examples in the EXAMPLE block

**Comment by chiehmin** [comment]
@Andersson007 Just added a changelog fragment and updated the example.

File: changelogs/fragments/461-resolve-domain-for-iscsi-portal.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+- allow open_iscsi module's portal field to be a domain name by resolving the portal ip address beforehand
```
Comment (Andersson007) [inline changelogs/fragments/461-resolve-domain-for-iscsi-portal.yml:2]:
```suggestion
- open_iscsi - allow ``portal`` parameter to be a domain name by resolving the portal ip address beforehand (https://github.com/ansible-collections/community.general/pull/461).
```

**Comment by Andersson007** [comment]
@srvg as the module's author, could you please take a look?

**Comment by Andersson007** [comment]
I just emailed the author. Let's wait a bit.

**Comment by srgvg** [comment]
It has been 7 years since I wrote this. I currently have no means to test the code, but looking at the changeset, this LGTM.

I assume
```
portal = socket.getaddrinfo(portal, None)[0][4][0]
```
just returns the same ip address when portal is set to an ip?

Thanks,

**Comment by Andersson007** [comment]
@chiehmin ^. @srgvg i examined the function, yes, it should return ip whatever (an ip addr or domain name) we pass

**Comment by srgvg** [comment]
shipit

**Comment by Andersson007** [comment]
@chiehmin thanks for the contribution!
@srgvg thanks for reviewing!

**Comment by Andersson007** [comment]
merged https://github.com/ansible-collections/community.general/pull/461 into master

**Pull request merged**